Framework/API/inc/MantidAPI/IFunction.h

```diff
@@ -583,6 +582,10 @@ class MANTID_API_DLL IFunction {
   Kernel::ProgressBase *m_progReporter;
 
 private:
+  /// No copying
+  IFunction(const IFunction&) = delete;
```

Comment:
The preferred use of `= delete` is to put the declarations in the public section so that the compiler generates a better error message: http://stackoverflow.com/questions/18931133/must-a-deleted-constructor-be-private.

Can we move these?